R/paths.R

```diff
@@ -453,3 +453,25 @@ del_empty_dir = function(dir) {
   files = list.files(dir, all.files = TRUE, no.. = TRUE)
   if (length(files) == 0) unlink(dir, recursive = TRUE)
 }
+
+#' Mark some paths as directories
+#'
+#' Add a trailing backlash to a file path if this is a directory. This is useful
+#' in messages to the console for example to quickly identify directories from
+#' files.
+#'
+#' If \code{files} is a vector of relative paths, directory test is done with
+#' path relative to the current working dir. Use \code{xfun::\link{in_dir}()} or
+#' use absolute path.
+#'
+#' @param files Character vector of paths to files and directories.
+#' @param dir directory from which \code{files} is relative to. Default to
+#'   current working directory.
+#' @examples
+#' mark_dirs(list.files(find.package("xfun"), full.names = TRUE))
+#' @export
+mark_dirs = function(files) {
+  i = dir_exists(files) & !grepl("/$", files)
+  files[i] = paste0(files[i], "/")
+  files
```

Comment:
I tend to just name the argument `x` like other functions such as `xfun::file_exists()`, since this function only has one argument, which should be relatively clear to document. The name `files` may indicate that the function only accept file paths but it actually also accepts dir paths.

```suggestion
mark_dirs = function(x) {
  i = dir_exists(x) & !grepl("/$", x)
  x[i] = paste0(x[i], "/")
  x
```